src/comment.ts

```diff
@@ -100,7 +102,10 @@ ${title}
 const getCommentId = (api: Gitlab, mrIid: number | string) =>
   api.MergeRequestNotes.all(context.projectId, mrIid).then(comments => {
     const changesetBotComment = comments.find(
-      comment => comment.author.username === process.env.GITLAB_CI_USER_NAME,
+      comment =>
+        comment.author.username === process.env.GITLAB_CI_USER_NAME &&
+        !comment.system &&
+        comment.body.includes('Generated By Changesets Bot'),
```

Comment:
Why this one is needed, personally I think a separate bot account should be used for such things, otherwise it's not same enough.

Reply:
This is the major bug that I was talking about. We were previously checking for username match only. That was not enough in case of the Gitlab. As GitLab creates `system` notes when you push new changes. You accidentally find that notes and try to update them.  
The last condition that you are talking about ensures that if the MR author added a note we don't think it is changesets-gitlab note. (I checked the API, this is the only way to distinguish these)

Reply:
In my imagination, a separate bot for changesets/CI things should be used to limit permissions, like changesets' bot account in its GitHub Action.

Is there any downside for this strategy?

Reply:
I think I don't understand your point here. Could you please elaborate more on this?  
We add a comment to MRs and without this change, the comment operation would be erroneous (It would throw 403 forbidden error or it would override one of the user's (non-changeset) comments).

Reply:
Why not use a separate bot account for comments like https://github.com/apps/changeset-bot? So that there will be no permission or conflict issue at all, right?

Reply:
Of course, the logic looks good to me, if you add some code comments here to describe why we're different from upstream here, that would be great.

For example, we don't have bot app like changesets-bot for all users.